### ISSUE TITLE
CMake 3.18+ build errors when building OpenAL backend

### ISSUE THREAD

**Issue by darktohka** [panda3d/panda3d#1330]
## Description

When building Panda3D with a CMake version higher than 3.18 and with the OpenAL backend enabled, Panda3D refuses to build:

```
In file included from /home/user/Desktop/panda3d/panda/src/audiotraits/config_openalAudio.cxx:16:
/home/user/Desktop/panda3d/panda/src/audiotraits/openalAudioManager.h:30:12: fatal error: 'AL/al.h' file not found
  #include <AL/al.h>
           ^~~~~~~~~
1 error generated.
```

This is because the `FindOpenAL.cmake` script, bundled with CMake, changed in CMake 3.18 and searches for "al.h", rather than "AL/al.h". As such, the include directory begins "openal/include/AL" rather than "openal/include", making it impossible for the compiler to find AL/al.h.

[`FindOpenAL.cmake`](https://cmake.org/cmake/help/v3.24/module/FindOpenAL.html) has the following comment on this:

```
Finds Open Audio Library (OpenAL).

Projects using this module should use ``#include "al.h"`` to include the OpenAL
header file, **not** ``#include <AL/al.h>``.  The reason for this is that the
latter is not entirely portable.  Windows/Creative Labs does not by default put
their headers in ``AL/`` and macOS uses the convention ``<OpenAL/al.h>``.
```

The `<AL/al.h>` and `<AL/alc.h>` includes would have to be replaced with `"al.h"` and `"alc.h"` includes. However, this would certainly break `makepanda` based projects, which still rely on `<AL/al.h>`. A unified solution has to be found.

## Steps to Reproduce

1. Install OpenAL development headers and libraries.
2. Clone the repository.
3. Build with CMake.
4. Wait for the error to appear.

## Environment
 * Operating system: Arch Linux
 * System architecture: x86_64
 * Panda3D version: 1.11.0 (master)
 * Installation method: built from source
 * Python version (if using Python): 3.10.5
 * Compiler (if using C++): clang 14.0.6

**Comment by rdb** [comment]
This is annoying.  All the documentation and example code indicates we have to use `<AL/al.h>` includes (and `<OpenAL/al.h>` for the macOS framework).  It would be annoying if we have to use a third alternative.

We can either just use the CMake convention in makepanda and change all our includes or we can ship our own FindOpenAL.cmake script.  I'm leaning towards the latter but I'll accept either PR.

**Comment by Germanunkol** [comment]
Just stumbled across the same issue.

If I read the linked documentation of FindOpenAL correctly then the change actually makes things _simpler_ because it removes the need to differentiate between OpenAL/al.h and AL/al.h (etc.) because cmake sets the include directory correctly, i.e. I think the directory will already include OpenAL on Mac (just as it does AL on linux).
So in the code this:
```
// OSX uses the OpenAL framework
#ifdef HAVE_OPENAL_FRAMEWORK
  #include <OpenAL/al.h>
  #include <OpenAL/alc.h>
#else
  #include <AL/al.h>
  #include <AL/alc.h>
  #include <AL/alext.h>
#endif
```
would just become the much cleaner:
```
  #include <al.h>
  #include <alc.h>
  #include <alext.h>
```

I was just working on a pull request but I don't understand what changes to make to makepanda.py (or makepanda.bat).
Is this change enough? But does this cover the MacOS case?

        SmartPkgEnable("OPENAL",   "openal",    ("openal"), "AL/al.h", framework = "OpenAL")

to 

        SmartPkgEnable("OPENAL",   "openal",    ("openal"), "al.h", framework = "OpenAL")

**Comment by rdb** [comment]
I don't agree with the proposed change because this is not how OpenAL wants to be included. We should follow upstream for the include conventions, not CMake.

We can just ship a modified copy of FindOpenAL.cmake.

**Comment by Moguri** [comment]
By the way, #1599  already includes a fix for this:


https://github.com/panda3d/panda3d/pull/1599/commits/e5b43ff980d34c429eb0bf0307fd3633bfa41abc
